activesupport/lib/active_support/testing/parallelize_executor.rb

```diff
@@ -45,19 +45,34 @@ def build_parallel_executor
 
         def parallelize
           @parallelized = true
-          Minitest::Test.parallelize_me!
+          parallel_test_suites.map(&:parallelize_me!)
         end
 
         def parallelized?
           @parallelized if defined?(@parallelized)
         end
 
         def should_parallelize?
-          ENV["PARALLEL_WORKERS"] || tests_count > threshold
+          ENV["PARALLEL_WORKERS"] || tests_count[:parallel] > threshold
+        end
+
+        def parallel_test_suites
+          Minitest::Runnable.runnables.select { |runnable| runnable.test_order == :parallel && !runnable.runnable_methods.empty? }
```

Comment:
This is inspired from how Minitest determines which tests suites should be run in parallel: [code](https://github.com/seattlerb/minitest/blob/master/lib/minitest.rb#L161)